packages/frontend/core/src/modules/tag/service/tag.ts

```diff
@@ -0,0 +1,72 @@
+import { LiveData, type PageRecordList } from '@toeverything/infra';
+import { nanoid } from 'nanoid';
+
+import type { WorkspaceLegacyProperties } from '../../workspace';
+import { Tag } from '../entities/tag';
+
+export class TagService {
+  constructor(
+    private readonly properties: WorkspaceLegacyProperties,
+    private readonly pageRecordList: PageRecordList
+  ) {}
+
+  readonly tags = this.properties.tagOptions$.map(tags =>
```

Comment:
Have we agreed upon livedata var/field naming? @EYHN

Reply:
Yes, I will rename them in later pr.